cloud/EAP7-1457_configMap_at_arbitrary_locations.adoc

```diff
@@ -0,0 +1,79 @@
+= Allow to mount ConfigMap at arbitrary locations from the operator
+:author:            Yeray Borges
+:email:             yborgess@redhat.com
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+ConfigMaps allow you to decouple configuration artifacts from image content to keep containerized applications portable. The Operator mounts the config maps as a volumes under the following path `/etc/configmaps/<config map name>`. This path currently is not configurable, which makes impossible to use the extension mechanism available in the server images. One of the use case of this extension mechanism is to execute any arbitrary bash script before / after the server configuration at runtime which requires the capability to mount a config map containing the bash scripts under `$JBOSS_HOME/extensions`.
+
+The goal of this feature is to allow the users to configure the mount path where the config map should be mounted.
+
+== Issue Metadata
+
+=== Issue
+
+* https://github.com/wildfly/wildfly-operator/issues/135[wildfly-operator/issues/135]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1457[EAP7-1457] - Allow to mount ConfigMap at arbitrary locations from the operator
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+ * mailto:
+
+=== Testing By
+
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+ * https://github.com/wildfly/wildfly-operator[WildFly Operator]
+
+=== Other Interested Projects
+
+== Requirements
+
+
+=== Hard Requirements
+
+Provide the ability to the user to specify not only the ConfigMap name, but also the path on the image filesystem where this config map should be mounted.
+
+* The specification of the mount path should be optional.
+* If a config map does not specify the mount path, the config map will be mounted under `/etc/configmaps/<config map name>`
+* Users should be able to specify absolute or relative paths. If it is a relative path, the path will be treated as relative to `JBOSS_HOME`.
```

Comment:
@yersan in Bootbale Jar images there is no `JBOSS_HOME`:

```
sh-4.4$ env | grep JBOSS
JBOSS_IMAGE_NAME=ubi8/openjdk-11
JBOSS_CONTAINER_MAVEN_DEFAULT_MODULE=/opt/jboss/container/maven/default/
JBOSS_CONTAINER_JAVA_RUN_MODULE=/opt/jboss/container/java/run
JBOSS_CONTAINER_MAVEN_36_MODULE=/opt/jboss/container/maven/36/
JBOSS_CONTAINER_S2I_CORE_MODULE=/opt/jboss/container/s2i/core/
JBOSS_CONTAINER_OPENJDK_JDK_MODULE=/opt/jboss/container/openjdk/jdk
JBOSS_CONTAINER_JOLOKIA_MODULE=/opt/jboss/container/jolokia
JBOSS_CONTAINER_JAVA_S2I_MODULE=/opt/jboss/container/java/s2i
JBOSS_CONTAINER_JAVA_JVM_MODULE=/opt/jboss/container/java/jvm
JBOSS_IMAGE_VERSION=1.3
JBOSS_CONTAINER_PROMETHEUS_MODULE=/opt/jboss/container/prometheus
JBOSS_CONTAINER_MAVEN_S2I_MODULE=/opt/jboss/container/maven/s2i
JBOSS_CONTAINER_UTIL_LOGGING_MODULE=/opt/jboss/container/util/logging/
JBOSS_CONTAINER_JAVA_PROXY_MODULE=/opt/jboss/container/java/proxy
```

hence the mechanism used to run customs script in the `$JBOSS_HOME/extensions` folder would not work for bootable jars; that's probably expected since the procedure to run `cli` for bootable jars is different: https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.3/html/using_eclipse_microprofile_with_jboss_eap_xp_2.0.0/the-bootable-jar_default#cli-scripts_default;

but, since the operator is supposed to work with both bootable and non bootable jar images, mentioning `JBOSS_HOME` as default for relative paths, leaves out the bootable jar image

Reply:
@yersan I’d like to revisit this topic. What will the relative path for images be when using a bootable jar?

Reply:
@mnovak1 When the Operator deploys a Bootable JAR application image, the image is extracted in a specific directory that acts as a JBOSS_HOME of that image. For the WildFly Operator, this path is `/opt/jboss/container/wildfly-bootable-jar-server`. We are in an early stage yet, but I guess additional considerations will need to be done for Bootable JAR images.